cmd/root.go

```diff
@@ -222,6 +229,116 @@ func preExecHook(cmd *cobra.Command, args []string) {
 			log.Fatalf("fsoc is not configured, please use \"fsoc config set\" to configure an initial context")
 		}
 	}
+	if time.Now().Unix()-getRecentTimestamp() > 86000 {
+		checkForUpdate()
+	}
+}
+
+func getRecentTimestamp() int64 {
+	potentialFiles, err := findFiles(os.TempDir())
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	var largestTimestamp int64 = 0
+	for i := 0; i < len(potentialFiles); i++ {
+		stringSections := strings.Split(potentialFiles[i].Name(), "/")
+		desiredSections := strings.Split(stringSections[len(stringSections)-1], "fsoctimestamp")
+		newTimestamp, _ := strconv.ParseInt(desiredSections[0], 10, 64)
+		if largestTimestamp < newTimestamp {
+			largestTimestamp = newTimestamp
+		}
+	}
+
+	for i := 0; i < len(potentialFiles); i++ {
+		err := os.Remove(potentialFiles[i].Name())
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+	}
+
+	return largestTimestamp
+}
+
+func findFiles(root string) ([]*os.File, error) {
+	var files []*os.File
+	err := filepath.WalkDir(root, func(pathh string, d fs.DirEntry, err error) error {
+		if !d.IsDir() {
+			if ok := strings.Contains(pathh, "fsoctimestamp"); ok && err == nil {
+				file, err := os.Open(pathh)
+				if err != nil {
+					log.Fatalf(err.Error())
+				}
+				files = append(files, file)
+			}
+		}
+		return nil
+	})
+	return files, err
+}
+
+func checkForUpdate() {
+	newestVersionChan := make(chan string)
+	go func() {
+		err := getVersion(newestVersionChan)
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+	}()
+	var newestVersion = <-newestVersionChan
+	if compareVersion(newestVersion) {
+		log.Warnf("There is a newer version of FSOC available, please upgrade to version %s", newestVersion)
```

Comment:
it may be nice to print both the current and the latest version. See pip, for example here https://stackoverflow.com/questions/46288847/how-to-suppress-pip-upgrade-warning
(although a single-line message is better; consider also using WithFields, so that the message can be parsed programmatically and the version extracted)